### ISSUE TITLE
[Helm] galley pod crash when using namespace other than istio-system

### ISSUE THREAD

**Issue by sebastien-prudhomme** [istio/istio#7446]
**Describe the bug**

With Helm chart of release-1.0 branch, if i use a namespace other than istio-system, the galley pod keep crashing:

```
2018-07-27T12:52:09.675481Z	warn	Could not find istio-system/istio-galley deployment to set ownerRef. The validatingwebhookconfiguration must be deleted manually
2018-07-27T12:52:09.675518Z	info	server-side configuration validation enabled
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x121d621]

goroutine 1 [running]:
istio.io/istio/galley/pkg/crd/validation.(*Webhook).rebuildWebhookConfig(0xc4202ffb00, 0x2c, 0x0)
	/workspace/go/src/istio.io/istio/galley/pkg/crd/validation/config.go:89 +0x271
istio.io/istio/galley/pkg/crd/validation.NewWebhook(0x1cf6640, 0xc4202435a0, 0x28db9a0, 0xe, 0xe, 0x0, 0x0, 0x1bb, 0x7ffc2bb9103d, 0x1f, ...)
	/workspace/go/src/istio.io/istio/galley/pkg/crd/validation/webhook.go:205 +0xabf
istio.io/istio/galley/cmd/galley/cmd.validatorCmd.func1(0xc4201de500, 0xc4200baa00, 0x0, 0x5)
	/workspace/go/src/istio.io/istio/galley/cmd/galley/cmd/validator.go:138 +0x498
istio.io/istio/vendor/github.com/spf13/cobra.(*Command).execute(0xc4201de500, 0xc4200ba960, 0x5, 0x5, 0xc4201de500, 0xc4200ba960)
	/workspace/go/src/istio.io/istio/vendor/github.com/spf13/cobra/command.go:760 +0x2c1
istio.io/istio/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4201de000, 0x0, 0x1b52e28, 0x13)
	/workspace/go/src/istio.io/istio/vendor/github.com/spf13/cobra/command.go:846 +0x30a
istio.io/istio/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4201de000, 0x6, 0x6)
	/workspace/go/src/istio.io/istio/vendor/github.com/spf13/cobra/command.go:794 +0x2b
main.main()
	/workspace/go/src/istio.io/istio/galley/cmd/galley/main.go:27 +0x8d
```

Seems that the chart is missing the option "--deployment-namespace={{ .Release.Namespace }}"

**Comment by sebastien-prudhomme** [comment]
Seems related to #7234

**Comment by sebastien-prudhomme** [comment]
I can confirm that adding the right option corrects the warn message but not the crash.

The crash happens because i set global.configValidation to false in the Helm chart. It seems that is not possible to disable the container --webhook-config-file option.

The help says:

```
      --webhook-config-file string     File that contains k8s validatingwebhookconfiguration yaml. Validation is disabled if file is not specified (default "/etc/istio/config")
```

How to not specified a file for this option?

**Comment by ayj** [comment]
`global.configValidation=false` should be sufficient to disable validation. If you're deploying galley in another namespace (other than istio-system) you'll need to make sure the `--deployment-namespace` flag for the Galley deployment is set correctly.

**Comment by sebastien-prudhomme** [comment]
Ok, resolved since PR#7477
